.github/workflows/update-and-rebuild.yaml

```diff
@@ -1,60 +1,116 @@
+          TENDERMINT_VERSION=$(grep 'github.com/tendermint/tendermint' go.mod | grep -v '=>' | awk '{print $2}' | awk -F '-' '{if ($3 != "") {print $3} else  if ($2 != "") {print $1"-"$2} else {print $1}}')
+          echo "NEW_TAG=${NEW_TAG}" >> $GITHUB_OUTPUT
+          echo "FINSCHIA_SDK_VERSION=$FINSCHIA_SDK_VERSION" >> $GITHUB_OUTPUT
+          echo "WASMD_VERSION=$WASMD_VERSION" >> $GITHUB_OUTPUT
+          echo "TENDERMINT_VERSION=$TENDERMINT_VERSION" >> $GITHUB_OUTPUT
+          echo "IBC_GO_VERSION=$IBC_GO_VERSION" >> $GITHUB_OUTPUT
+
+  create-bump-up-PR:
+    runs-on: ubuntu-latest
+    needs: check-new-finschia-tag
+    if: needs.check-new-finschia-tag.outputs.NEW_TAG
+    env:
+      NEW_TAG: ${{ needs.check-new-finschia-tag.outputs.NEW_TAG }}
+      FINSCHIA_SDK_VERSION: ${{ needs.check-new-finschia-tag.outputs.FINSCHIA_SDK_VERSION }}
+      WASMD_VERSION: ${{ needs.check-new-finschia-tag.outputs.WASMD_VERSION }}
+      IBC_GO_VERSION: ${{ needs.check-new-finschia-tag.outputs.IBC_GO_VERSION }}
+      TENDERMINT_VERSION: ${{ needs.check-new-finschia-tag.outputs.TENDERMINT_VERSION }}
+
+    steps:
+      - name: Check out to finschia-wasm
+        uses: actions/checkout@v4
+        with:
+          repository: Finschia/finschia-wasm
+
+      - name: fetch and merge submodule
+        run: |
+          git submodule update --init --remote
+          git submodule status
 
-      - name: Checkout
-        uses: actions/checkout@v3
+      - name: rust setup
+        run: rustup component add rustfmt
 
-      - name: Set up cargo cache
-        uses: actions/cache@v3
-        continue-on-error: false
+      - name: buf setup
+        uses: bufbuild/buf-setup-action@v1
+
+      - name: Install protoc
+        uses: arduino/setup-protoc@v1
         with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: ${{ runner.os }}-cargo-
-
-      - name: Generate code
+          version: "3.14.0"
+
+      - name: Install buf
         run: |
-          git config --global user.name "github-actions[bot]"
-          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          GO111MODULE=on go install github.com/bufbuild/buf/cmd/buf@v1.15.1
 
-          # buf is installed in GOPATH
-          export PATH=${PATH}:`go env GOPATH`/bin
+      - name: use actions cache
+        uses: ./.github/actions/cache
 
-          ./scripts/update-and-rebuild.sh "$OSMOSIS_REV"
+      - name: set each submodule version to env file in finschia-wasm
+        run: |
+          echo '# Choose from https://github.com/Finschia/finschia/releases' | cat > env
```

Comment:
isn't it enough to use `> env` instead of `| cat > env`?

Reply:
Thanks. Fixed here.
[fix: delete unnecessary "| cat" when the env file is overwritten](https://github.com/Finschia/finschia-wasm/pull/9/commits/7895f81bdc181a0443311d3b4f338cb6be3b6702)